src/tracks/gosling-track/gosling-track.ts

```diff
@@ -1472,6 +1501,31 @@ const factory: PluginTrackFactory<Tile, GoslingTrackOptions> = (HGC, context, op
                 }
             });
         }
+
+        /**
+         * Used in drawTile()
+         * Checks if the track has marks which are stretchable. Stretching
+         * is not supported for circular layouts or 2D tracks
+         */
+        #hasStretchableGraphics() {
+            const stretchableMarks = ['bar', 'line', 'rect', 'area'];
+            return (
+                this.options.spec.experimental?.stretchGraphics ||
+                (!Is2DTrack(this.getResolvedTracks()[0]) &&
+                    this.options.spec.layout !== 'circular' &&
+                    stretchableMarks.includes(this.options.spec.mark || ''))
+            );
+        }
+
+        /**
+         * Used in drawTile()
+         * Checks if the tile Graphic is too stretched. If so, it returns true.
+         * @param stretchFactor The factor by which the tile is stretched
+         * @returns True if the tile is too stretched, false otherwise
+         */
+        #isTooStretched(stretchFactor: number) {
+            return stretchFactor > 2 || stretchFactor < 0.5;
```

Comment:
Could we add another experimental option for controlling the threshold? I would like to experiment with this using some examples.

```suggestion
            const threshold = this.options.spec.experimental?.stretchGraphicsThreshold ?? 2;
            return stretchFactor > threshold || stretchFactor < 1 / threshold;
```

Reply:
That's a good idea!